### DISCUSSION TITLE
Custom JSON Schema and Automatic TS Interface Generation for JSON and Lexical RTE Fields

### DISCUSSION THREAD

**Discussion by PatrikKozak** [payloadcms/payload#4879] [Feature Requests & Ideas]
Currently, both `JSON` fields and `Lexical Rich Text Editor` (RTE) fields are typed generically within the generated types. 

### Enhancement:

For JSON fields, allow users to define their own JSON schema and, based on it, automatically generate a TypeScript interface for the field. 

For Lexical Rich Text Editor (RTE) fields, automatically construct TypeScript types for the RTE fields based on the enabled nodes.

**Comment by JarrodMFlesch** [comment]
Added in [this PR](https://github.com/payloadcms/payload/pull/5898).
